### PR TITLE
Merged tschaub/mock-fs into Slayer95/mock-fs for getting Node 8 support

### PR DESCRIPTION
**Pull request by webextensions**

### CODE REVIEW

File: test/lib/index.spec.js [hunk 1]
```diff
@@ -65,7 +75,7 @@ describe('The API', function() {
       mock.restore();
     });
 
-    it('uses the real fs module in require() calls', function() {
+    xit('uses the real fs module in require() calls', function() {
```
Comment (Slayer95) [inline test/lib/index.spec.js:78]: The whole point of this fork is that this test should pass :|

**Comment by webextensions** [comment]
My bad ... didn't realize that part in the merge ... the tests are erroring out with Node 8, so attempted if this could be a quickfix.

Like #203, would it be possible for you to extend support to Node 8 ?

**Comment by Slayer95** [comment]
I definitely want to extend support to Node 8.

However, there have been plenty of non-trivial changes in the Node.js internals, specially regarding ``stat``. Therefore, I must have a deeper look at the problem and decide whether to stick with the ``mock-fs@3`` approach of copying the ``fs`` sources -with appropriate edits-, or rebase on ``mock-fs@4`` and reimplement module-load hooks to ensure the right behavior with ``require``.

I am currently leaning towards the second option. However, working out the details is not my top priority right now.

**Comment by webextensions** [comment]
Thanks for considering ... Closing my invalid PR :-)

**Pull request closed without merging**